core/services/PolicyService.js

```diff
@@ -1,10 +1,22 @@
+const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));
+
 module.exports = {
-  checkAndInjectOperator: (ctx, next) => {
+  checkAndInjectOperator: async (ctx, next) => {
     try {
       ctx.operator = AuthService.checkAndDecodeToken(ctx.cookies.get("jwt"));
-      next();
+      await next();
     } catch (err) {
       ctx.throw(401, "unauthorized");
     }
+  },
+
+  hasRight: (modelKey, right) => {
```

Comment:
All what your `hasRight ` function does is just returning some expression. In such cases you should use short function form, without curly brackets. Speaking about the function being returned, we also can apply this approach, so the whole code can take 4 lines instead of 8:
```
hasRight: (modelKey, right) => async (ctx, next) => 
  AuthService.checkRight(ctx.operator, modelKey, right)
    ? await next();
    : ctx.throw(403, "You don't have rights to perform this action")
```